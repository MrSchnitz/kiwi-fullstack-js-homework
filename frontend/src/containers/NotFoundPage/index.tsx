import React from "react";
import styled from "styled-components/macro";
import {withNavigation} from "../../hoc/NavBar";

function NotFoundPage() {
  return (
    <>
      <Wrapper>
        <Title>
          4
          <span role="img" aria-label="Crying Face">
            ðŸ˜¢
          </span>
          4
        </Title>
        <p>Page not found.</p>
      </Wrapper>
    </>
  );
}

const Wrapper = styled.div`
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  min-height: 320px;
`;

const Title = styled.div`
  margin-top: -8vh;
  font-weight: bold;
  color: black;
  font-size: 3.375rem;

  span {
    font-size: 3.125rem;
  }
`;

export default withNavigation(NotFoundPage);
