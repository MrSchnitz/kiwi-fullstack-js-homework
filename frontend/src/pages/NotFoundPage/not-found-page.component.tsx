import React from "react";
import {NotFoundPageLayout, Title} from "./not-found-page.styles";

function NotFoundPage() {
  return (
    <>
      <NotFoundPageLayout>
        <Title>
          4
          <span role="img" aria-label="Crying Face">
            ðŸ˜¬
          </span>
          4
          <p>Page not found.</p>
        </Title>
      </NotFoundPageLayout>
    </>
  );
}

export default NotFoundPage;
