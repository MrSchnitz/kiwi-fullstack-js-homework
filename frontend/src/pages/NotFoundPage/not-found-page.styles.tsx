import styled from "styled-components/macro";

export const NotFoundPageLayout = styled.div`
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export const Title = styled.div`
  font-weight: bold;
  color: black;
  font-size: 6rem;
  text-align: center;

  span {
    font-size: 5.7rem;
  }

  p {
    font-weight: normal;
    font-size: 2.5rem;
  }
`;
