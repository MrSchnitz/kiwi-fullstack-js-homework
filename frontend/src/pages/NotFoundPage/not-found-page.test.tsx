import React from "react";
import {shallow, ShallowWrapper} from "enzyme";
import NotFoundPage from "./not-found-page.component";

describe("NotFound page", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<NotFoundPage />);
  });

  it("should render NotFound page", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
