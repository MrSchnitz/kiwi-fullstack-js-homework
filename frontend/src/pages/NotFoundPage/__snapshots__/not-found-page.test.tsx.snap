// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page should render NotFound page 1`] = `ShallowWrapper {}`;
